packages/main/src/Menu.ts

```diff
@@ -341,27 +349,45 @@ class Menu extends UI5Element {
 	}
 
 	_itemKeyDown(e: KeyboardEvent) {
-		if (!isLeft(e) && !isRight(e)) {
-			return;
-		}
-
-		const shouldCloseMenu = this.isRtl ? isRight(e) : isLeft(e);
-		const shouldOpenMenu = this.isRtl ? isLeft(e) : isRight(e);
+		const isTabNextPrevious = isTabNext(e) || isTabPrevious(e);
 		const item = e.target as MenuItem;
 		const parentElement = item.parentElement as MenuItem;
+		const shouldCloseMenu = parentElement.hasAttribute("ui5-menu-item") && (this.isRtl ? isRight(e) : isLeft(e));
+		const shouldOpenMenu = this.isRtl ? isLeft(e) : isRight(e);
```

Comment:
Move this line above the previous line and reuse it:

```ts
const shouldOpenMenu = this.isRtl ? isLeft(e) : isRight(e);
const shouldCloseMenu = !shouldOpenMenu && parentElement.hasAttribute("ui5-menu-item");
```

Also, when you have an && expression, you should put the faster condition first, and the slower second. In this case the `.hasAttribute` one is much slower because it queries the DOM.